src/framework/xr/xr-hit-test-source.js

```diff
@@ -132,23 +132,44 @@ class XrHitTestSource extends EventHandler {
      * @private
      */
     updateHitResults(results, inputSource) {
+        if (inputSource && !inputSource.hitTestSourcesSet.has(this))
+            return;
+
+        let origin = poolVec3.pop();
+        if (!origin) origin = new Vec3();
```

Comment:
```suggestion
        const origin = poolVec3.pop() ?? new Vec3();
```

Reply:
and the same can be done for position / rotation below

Reply:
Good tip!